privacyidea/lib/monitoringmodules/base.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+#
+#  2018-11-22 Initial create
+#             Cornelius Kölbel <cornelius.koelbel@netknights.it>
+#
+#  License:  AGPLv3
+#  contact:  http://www.privacyidea.org
+#
+# This code is free software; you can redistribute it and/or
+# modify it under the terms of the GNU AFFERO GENERAL PUBLIC LICENSE
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or any later version.
+#
+# This code is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU AFFERO GENERAL PUBLIC LICENSE for more details.
+#
+# You should have received a copy of the GNU Affero General Public
+# License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+__doc__ = """This is the BaseClass for monitoring statistics data.
+
+The monitoring database is a time series database. It is supposed to store
+tuples of (timestamp, status key, status value)
+
+A monitoring module needs to provide the possibility to write new data,
+return all available keys, return the last data. 
+"""
+import logging
+log = logging.getLogger(__name__)
+from privacyidea.lib.log import log_with
+
+
+class Monitoring(object):
+
+    def __init__(self, config=None):
+        pass
+
+    def get_keys(self):
+        """
+        Return a list of the available statistic keys.
+
+        :return: list of identifiers
+        """
+        return []
+
+    def get_values(self, stats_key, start_timestamp=None, end_timestamp=None, date_strings=False):
+        """
+        Return a list of tuples of (timestamp, value) for the requested stats_key.
```

Comment:
We should clarify that the timestamps are inclusive.

Reply:
really?

Reply:
:+1: